pkg/scheduler/apis/config/validation/validation_pluginargs.go

```diff
@@ -31,6 +31,12 @@ import (
 	"k8s.io/kubernetes/pkg/scheduler/apis/config"
 )
 
+var supportedScoringStrategyType = sets.NewString(
```

Comment:
`supportedScoringStrategyTypes` or `supportedScoringStrategySet` ?

Reply:
Define a new set here may grow the maintenance costs, like when adding a new type in api package, but considering that the amount of ScoringStrategy types is quite stable, I think it's ok here.

Reply:
This is a reference to ServiceType:
https://github.com/kubernetes/kubernetes/blob/eb2f56c638b393fd589f4bda7b7b84d4d2ccfbdf/pkg/apis/core/validation/validation.go#L4485-L4486

It is ok, if we want to use `supportedScoringStrategyTypes`.

Reply:
> Define a new set here may grow the maintenance costs, like when adding a new type in api package, but considering that the amount of ScoringStrategy types is quite stable, I think it's ok here.

If a new type is added and we forget to update the `supportedScoringStrategyTypes`, I think the scheduler will fail to start with an unsupported type and we can easily find the issue and fix it. Our e2e tests should catch this.

Reply:
Yes if we have tests cover this when adding a new type. I mentioned this here for you used to fetch the supported types from `nodeResourceStrategyTypeMap`, now you defined a new variable here. I think it may because of the package circular dependency, right? And I think it's ok here.

Reply:
Yes, Thanks!